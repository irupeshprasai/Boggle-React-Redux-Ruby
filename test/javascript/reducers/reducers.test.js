import reducer from '../../../app/javascript/store/reducers/game.reducer'
import { initialState } from '../../../app/javascript/store/reducers/game.reducer'

import {
    NEW_GAME_INIT, NEW_GAME_SUCCESS, NEW_GAME_FAILURE,
    EVALUATION_INIT,
    EVALUATION_SUCCESS,
    EVALUATION_INCORRECT,
    EVALUATION_FAILURE,
    CLEAR_GAME
} from '../../../app/javascript/store/types'
import { GAMEMOCKS } from '../__mocks__/gameMocks';
describe('game reducer', () => {
    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot()
    })

    it('should handle a new Game init request', () => {
        expect(
            reducer(initialState,
                { type: NEW_GAME_INIT, user: GAMEMOCKS.GAME_INIT.user })
        ).toMatchSnapshot()
    });

    it('should init a new game', () => {
        expect(
            reducer(initialState,
                { type: NEW_GAME_SUCCESS, user: GAMEMOCKS.GAME_INIT_SUCCESS.user })
        ).toMatchSnapshot()
    });

    it('should handle game failures', () => {
        expect(
            reducer(initialState,
                { type: NEW_GAME_FAILURE })
        ).toMatchSnapshot()
    });

    it('should handle evaluation init', () => {
        expect(
            reducer(initialState,
                { type: EVALUATION_INIT, response: GAMEMOCKS.EVALUATION_INIT.response })
        ).toMatchSnapshot()
    });


    it('should handle evaluation sucess', () => {
        expect(
            reducer(initialState,
                { type: EVALUATION_SUCCESS, response: GAMEMOCKS.EVALUATION_SUCCESS.response })
        ).toMatchSnapshot()
    });

    it('should handle incorrect evaluations.', () => {
        expect(
            reducer(initialState,
                {
                    type: EVALUATION_INCORRECT
                })
        ).toMatchSnapshot()
    });

    it('should handle evaluation failures.', () => {
        expect(
            reducer(initialState,
                {
                    type: EVALUATION_FAILURE
                })
        ).toMatchSnapshot()
    });


    it('should handle clearing the game.', () => {
        expect(
            reducer(initialState,
                {
                    type: CLEAR_GAME
                })
        ).toMatchSnapshot()
    });

})