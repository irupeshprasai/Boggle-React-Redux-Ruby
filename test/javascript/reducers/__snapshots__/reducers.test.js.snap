// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game reducer should handle a new Game init request 1`] = `
Object {
  "baseName": "/",
  "currentUser": Object {
    "stageID": 1,
    "userName": "Ozesh",
  },
  "defaultPath": "/",
  "errMsg": Array [],
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": true,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;

exports[`game reducer should handle clearing the game. 1`] = `
Object {
  "baseName": "/",
  "currentUser": null,
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": false,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;

exports[`game reducer should handle evaluation failures. 1`] = `
Object {
  "baseName": "/",
  "currentUser": null,
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": false,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;

exports[`game reducer should handle evaluation init 1`] = `
Object {
  "baseName": "/",
  "currentUser": null,
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": false,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 1,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;

exports[`game reducer should handle evaluation sucess 1`] = `
Object {
  "baseName": "/",
  "currentUser": null,
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": false,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {
    "toad": 1,
  },
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;

exports[`game reducer should handle game failures 1`] = `
Object {
  "baseName": "/",
  "currentUser": null,
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": false,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;

exports[`game reducer should handle incorrect evaluations. 1`] = `
Object {
  "baseName": "/",
  "currentUser": null,
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": false,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 1,
}
`;

exports[`game reducer should init a new game 1`] = `
Object {
  "baseName": "/",
  "currentUser": Object {
    "boardSize": 4,
    "data": Object {
      "board_data": "dtyraqtaoarexepm",
    },
    "message": "New game initiated.",
    "stageID": 1,
    "success": true,
    "userName": "Ozesh",
  },
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": true,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;

exports[`game reducer should return the initial state 1`] = `
Object {
  "baseName": "/",
  "currentUser": null,
  "defaultPath": "/",
  "gamePlay": Object {
    "diagonal": true,
    "horizontal": true,
    "vertical": true,
  },
  "inGame": false,
  "minLetters": 3,
  "stages": Array [],
  "timeLimit": "00:03:00",
  "trials": 0,
  "users": Array [
    Object {
      "level": 3,
      "score": 34,
      "trials": 2,
      "user": "andrew",
    },
  ],
  "wordScoreList": Object {},
  "wordTypes": Object {
    "adjectives": true,
    "adverbs": true,
    "nouns": true,
    "properNouns": false,
    "verbs": true,
  },
  "wrongCount": 0,
}
`;
